src/validator.ts

```diff
@@ -45,14 +43,21 @@ export = function validator(
   // MAY support others.
   const defaultQueryLanguage = definition.QueryLanguage ?? "JSONPath";
 
-  const IsJsonPath = ({ state }: StateEntry): boolean => {
-    const queryLanguage = state.QueryLanguage ?? defaultQueryLanguage;
-    return queryLanguage === "JSONPath";
-  };
-  const IsJsonNata = ({ state }: StateEntry): boolean => {
-    const queryLanguage = state.QueryLanguage ?? defaultQueryLanguage;
-    return queryLanguage === "JSONata";
-  };
+  // Select all objects named "States"
+  const statesObjects: Record<string, unknown>[] = JSONPath({
+    path: "$..States",
+    json: definition,
+  });
+  for (const states of statesObjects) {
+    if (states && typeof states === "object") {
```

Comment:
What's the reason to check the truthyness and type (here and in the next condition)?

Doesn't JSONPath return something consistent?

```ts
  for (const states of statesObjects) {
    for (const stateName of Object.keys(states)) {
      const state = states[stateName] as { QueryLanguage?: string };
      if (!state.QueryLanguage) {
        state.QueryLanguage = defaultQueryLanguage;
      }
    }
```